app/src/main/java/org/greenstand/android/TreeTracker/models/Configuration.kt

```diff
@@ -0,0 +1,37 @@
+package org.greenstand.android.TreeTracker.models
+
+import com.google.gson.GsonBuilder
+import org.greenstand.android.TreeTracker.preferences.PrefKey
+import org.greenstand.android.TreeTracker.preferences.PrefKeys
+import org.greenstand.android.TreeTracker.preferences.Preferences
+
+class Configuration(private val preferences: Preferences) {
+
+    private val LOCATION_DATA_CONFIG_KEY: PrefKey = PrefKeys.SYSTEM_SETTINGS +
+            PrefKey("location-data-config")
+    private val gson = GsonBuilder().serializeNulls().create()
+    lateinit var locationDataConfig: LocationDataConfig
+        private set
+
+    init {
+        locationDataConfig = LocationDataConfig() // Initialize with default value
+        val locationDataConfigString = preferences.getString(LOCATION_DATA_CONFIG_KEY)

```

Comment:
There doesn't seem to be a nice for this local value.

Reply:
I am not sure what your observation is. What is happening here is `LocationDataConfig` takes the constructor default values for the config parameters and is overriden by the values from preferences if a value for `LOCATION_DATA_CONFIG_KEY` is found. Otherwise retains the default constructor values. Required the very first time the app is installed or when the user hasn't fiddled with the config parameters.

Reply:
Sorry, typo. Ignore this